src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -80,15 +89,19 @@ class FastSync(
         context become idle
         syncController ! Done
       } else {
-        val initialSyncState = SyncState(targetBlockHeader, pendingMptNodes = Seq(StateMptNodeHash(targetBlockHeader.stateRoot)))
+        val initialSyncState =
+          SyncState(targetBlockHeader, safeDownloadTarget = targetBlockHeader.number + syncConfig.fastSyncBlockValidationX)
         startWithState(initialSyncState)
       }
   }
 
+  // scalastyle:off number.of.methods
   private class SyncingHandler(initialSyncState: SyncState) {
 
     private val BlockHeadersHandlerName = "block-headers-request-handler"
 
+    private var requestedHeaders: Map[Peer, BigInt] = Map.empty
```

Comment:
Should we remove a peer from this map if a request failed (in the `handleRequestFailure` method)? As this map is checked when assigning work (in `assignBlockchainWork`) it seems it could be the case that headers are no longer requested if the map is not cleared.

Reply:
great catch 👍